src/pages/SwapNew/components/CurrencyItem/CurrencySymbol.tsx

```diff
@@ -0,0 +1,36 @@
+import { Currency } from '@swapr/sdk'
+
+import { useTranslation } from 'react-i18next'
+import styled from 'styled-components'
+
+import { TEXT_COLOR_PRIMARY } from '../../constants'
+import { FontFamily } from '../../constants'
+
+type CurrencySymbolProps = {
+  currency?: Currency | null
+}
+
+export function CurrencySymbol({ currency }: CurrencySymbolProps) {
+  const { t } = useTranslation('swap')
+
+  return (
+    <Paragraph>
+      {(currency && currency.symbol && currency.symbol.length > 20
```

Comment:
Why 20 can it be 12 based on the fallback of 4+3+5 characters
```suggestion
      {(currency?.symbol?.length > 12
```

Reply:
This is not based on the fallback, this is based on if the text can fit the swapbox. E.g. token symbol `GSWAPRTVL-0929` fits nicely in the swapbox so there's no reason to substring it. I tested this and updated to substring after 15 characters. This is also dependant on font.